src/inmanta/data/model.py

```diff
@@ -130,8 +130,8 @@ class AttributeStateChange(BaseModel):
     Changes in the attribute
     """
 
-    current: Optional[Union[SimpleTypes, List[SimpleTypes], Dict[str, SimpleTypes]]] = None
-    desired: Optional[Union[SimpleTypes, List[SimpleTypes], Dict[str, SimpleTypes]]] = None
+    current: Optional[Any] = None
```

Comment:
These are also send to the server, so that means that now anything can go in?

Reply:
From my discussion with @wouterdb yesterday I understood that this object was only created via methods in handler.py. These methods already had `Any` in their definition, see the other change in this PR.

Reply:
They indeed are any, however they need to be something that we can send to the server. That is the reason why it was typed liked this.

Reply:
Yes, but we have no good super type for 'serializable' 

And to define such a type, which is both correct for mypy and works for pydantic would be very very difficult. 
Because pydantic has no 'hands-off' mode, it accepts the types as they are (except for the `Any` type). 

If we want to define such a super type, the union would have to be ordered in such a way that for any element E in the type union there is no valid instance of E that pydantic can convert to an object of any type prior to E in the union. (But because the inheritance hiërarchie is open to extension (new subtypes can be added any time, even at runtime) this property can not be proven/achieved.